Sources/PointFree/Newsletters.swift

```diff
@@ -1,13 +1,55 @@
 import ApplicativeRouter
+import Either
 import HttpPipeline
 import Foundation
 import Prelude
 import Tuple
 
 let expressUnsubscribeMiddleware =
-{ (conn: Conn<StatusLineOpen, T4<Database.User?, Database.User.Id, Database.EmailSetting.Newsletter, Prelude.Unit>>) -> IO<Conn<StatusLineOpen, Prelude.Unit>> in
+  unsubscribeMiddleware
+    >-> redirect(to: .secretHome)
 
-  let (_, userId, newsletter) = lower(conn.data)
+let expressUnsubscribeReplyMiddleware =
+  requireUserAndNewsletter
+    <| unsubscribeMiddleware
+    >-> writeStatus(.ok)
+    >-> respond(text: "OK")
```

Comment:
unsubscribing via reply needs to send back a `200` so that mailgun knows it doesnt need to try again

Reply:
Does it really need the "OK" body? 😄